pool.go

```diff
@@ -509,9 +285,16 @@ func (p *pool) fetchResourceWith(ctx context.Context, path string, cb DataCallba
 	}
 	p.fetchKeyLk.RUnlock()
 
-	nodes, err := p.getNodesToFetch(path, with)
-	if err != nil {
-		return err
+	aff := with
+	if aff == "" {
+		aff = path
+	}
+
+	p.lk.Lock()
```

Comment:
likewise: why is this an exclusive lock?

Reply:
Yeah agreed this can be just a read lock. Fixed.